File: src-tauri/src/commands/system_info.rs
use serde::Serialize;

use crate::system_info::{self, PlatformInfo};

#[derive(Debug, Clone, Serialize)]
pub struct SystemInfo {
    pub platform: PlatformInfo,
    pub cpu: String,
    pub memory: String,
    pub gpu: String,
    pub storage: String,
}

#[tauri::command]
pub fn get_system_info() -> Result<SystemInfo, String> {
    Ok(SystemInfo {
        platform: system_info::get_platform_info(),
        cpu: system_info::get_cpu_info(),
        memory: system_info::get_memory_info(),
        gpu: system_info::get_gpu_info(),
        storage: system_info::get_storage_info(),
    })
}

File: src/lib/queries/system-info.ts
import { createQuery } from "@tanstack/svelte-query";
import { invoke } from "@tauri-apps/api/core";

type PlatformInfo = {
  os: string;
  hostname: string;
};

type SystemInfo = {
  platform: PlatformInfo;
  cpu: string;
  memory: string;
  gpu: string;
  storage: string;
};

export function useSystemInfo() {
  return createQuery(() => ({
    queryKey: ["system-info"],
    queryFn: async () =>
      await invoke<SystemInfo>("get_system_info"),
    staleTime: 5 * 60 * 1000,
    gcTime: 10 * 60 * 1000,
    retry: 2,
  }));
}
